internal/olm/operator/internal/registry.go

```diff
@@ -0,0 +1,96 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"context"
+	"fmt"
+
+	olmclient "github.com/operator-framework/operator-sdk/internal/olm/client"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// SDKLabels are used to identify certain operator-sdk resources.
+var SDKLabels = map[string]string{
+	"owner": "operator-sdk",
+}
+
+// RegistryResources configures creation/deletion of internal registry-related
+// resources.
+type RegistryResources struct {
+	Client  *olmclient.Client
+	Pkg     registry.PackageManifest
+	Bundles []*registry.Bundle
+}
+
+// FEAT(estroz): allow users to specify labels for registry objects.
+
+// CreateRegistryManifests creates all registry objects required to serve
+// manifests from m.manifests in namespace.
+func (m *RegistryResources) CreateRegistryManifests(ctx context.Context, namespace string) error {
+	pkgName := m.Pkg.PackageName
+	binaryKeyValues, err := createConfigMapBinaryData(m.Pkg, m.Bundles)
+	if err != nil {
+		return fmt.Errorf("error creating registry ConfigMap binary data: %w", err)
+	}
+	cm := newRegistryConfigMap(pkgName, namespace,
+		withBinaryData(binaryKeyValues),
+	)
+	volName := getRegistryVolumeName(pkgName)
+	dep := newRegistryDeployment(pkgName, namespace,
+		withRegistryGRPCContainer(pkgName),
+		withVolumeConfigMap(volName, cm.GetName()),
+		withContainerVolumeMounts(volName, []string{containerManifestsDir}),
+	)
+	service := newRegistryService(pkgName, namespace,
+		withTCPPort("grpc", registryGRPCPort),
+	)
+	if err = m.Client.DoCreate(ctx, cm, dep, service); err != nil {
+		return fmt.Errorf("error creating operator %q registry-server objects: %w", pkgName, err)
+	}
+	depKey := types.NamespacedName{
+		Name:      dep.GetName(),
+		Namespace: namespace,
+	}
+	log.Infof("Waiting for Deployment %q rollout to complete", depKey)
+	if err = m.Client.DoRolloutWait(ctx, depKey); err != nil {
+		return fmt.Errorf("error waiting for Deployment %q to roll out: %w", depKey, err)
+	}
+	return nil
+}
+
+// DeleteRegistryManifests deletes all registry objects serving manifests
+// from m.manifests in namespace.
+func (m *RegistryResources) DeleteRegistryManifests(ctx context.Context, namespace string) error {
+	pkgName := m.Pkg.PackageName
+	cm := newRegistryConfigMap(pkgName, namespace)
+	dep := newRegistryDeployment(pkgName, namespace)
+	service := newRegistryService(pkgName, namespace)
```

Comment:
Why is required re-create the full objects to delete them? Why not get the object by the name and namespace and then, delete it? Could not this impl face pitfuls scenarios where the object in the cluster was updated and is not so longer exactly equals the generation? Will not it face an error if the object for some reason was not really created?

Reply:
`Client.Delete()` only cares about object metadata (type info, namespace, and name), which an empty object provides. `Client.Get()`-ing an object first would be an unnecessary network-bound operation because we know all necessary type information before calling `Client.Delete()`.

Reply:
IHMO: would be better:
// get the resource by name and namespace
// if exist remove and log
// if not exist just log

Reply:
:bulb: I now see what @camilamacedo86 is saying because the method names say `newXXX` it looks like you are getting back a FULL XXX object. Looking at the implementation of `newXXX` you see it creates metadata only since there was no `with` arguments specified. This one does seem a bit harder to maintain in the long run. It feels *too* clever :) What @camilamacedo86 stated would definitely be more *clear* to the maintainer of what is REALLY going on. 

I need to think about this one a little more.

Reply:
I do not see why having an extra `Get()` step would be beneficial, since all `Get()` is doing in this context is populating metadata; this is what `newXXX()` does implicitly, without a request. Perhaps a comment on this bit of could will suffice? Code comments on each `newXXX()` method explain that only metadata is populated in the returned object.